sorc/checkout.sh

```diff
@@ -131,14 +147,23 @@ mkdir -p ${logdir}
 # The checkout version should always be a speciifc commit (hash or tag), not a branch
 errs=0
 checkout "ufs_model.fd"    "https://github.com/ufs-community/ufs-weather-model" "${ufs_model_hash:-b97375c}"      ; errs=$((errs + $?))
-checkout "gsi_enkf.fd"     "https://github.com/NOAA-EMC/GSI.git"                "67f5ab4"                         ; errs=$((errs + $?))
-checkout "gsi_utils.fd"    "https://github.com/NOAA-EMC/GSI-Utils.git"          "322cc7b"                         ; errs=$((errs + $?))
-checkout "gsi_monitor.fd"  "https://github.com/NOAA-EMC/GSI-Monitor.git"        "acf8870"                         ; errs=$((errs + $?))
-checkout "gdas.cd"         "https://github.com/NOAA-EMC/GDASApp.git"            "5952c9d"                         ; errs=$((errs + $?))
-checkout "gldas.fd"        "https://github.com/NOAA-EMC/GLDAS.git"              "fd8ba62"                         ; errs=$((errs + $?))
 checkout "ufs_utils.fd"    "https://github.com/ufs-community/UFS_UTILS.git"     "a2b0817"                         ; errs=$((errs + $?))
 checkout "verif-global.fd" "https://github.com/NOAA-EMC/EMC_verif-global.git"   "c267780"                         ; errs=$((errs + $?))
 
+if [[ "${CHECKOUT_GSI:-NO}" == "YES" ]]; then
```

Comment:
Define `CHECKOUT_GSI=NO` and `CHECKOUT_GDAS=NO` at the top of `checkout.sh` before the arguments are parsed.

Reply:
_at the top of checkout.sh_  - Is this the literal top

```
#!/bin/sh
set +x
set -u

CHECKOUT_GSI=NO
CHECKOUT_GDAS=NO

function usage() {
```

or after the function definitions

```
  fi
  echo
  return 0
}

CHECKOUT_GSI=NO
CHECKOUT_GDAS=NO

while getopts ":chgum:o" option; do
  case $option in
    c)
```

If we set `CHECKOUT_GSI` and `CHECKOUT_GDAS` in either of these locations, the default definitions in if tests are no longer necessary

```
if [[ "${CHECKOUT_GSI:-NO}" == "YES" ]]; then
  checkout "gsi_enkf.fd" "https://github.com/NOAA-EMC/GSI.git" "67f5ab4"; errs=$((errs + $?))
fi

if [[ "${CHECKOUT_GDAS:-NO}" == "YES" ]]; then
  checkout "gdas.cd" "https://github.com/NOAA-EMC/GDASApp.git" "5952c9d"; errs=$((errs + $?))
fi
```

If we adopt this approach, shall we apply it to other variables:  `CLEAN`, `checkout_gtg`, `checkout_wafs`, `ufs_model_hash`?

Reply:
Commit @CoryMartin-NOAA 's suggestion at [ff2192d](https://github.com/NOAA-EMC/global-workflow/commit/ff2192dc6fa005ee52b8c8493271a27a8ef569c2)

Reply:
@RussTreadon-NOAA 
The latter is what I had in mind; i.e. after the function definitions.
And yes, if you would do the same for the other argument options, that would be great.  Thanks~

Reply:
Suggested change committed to `feature/clone` at [ced1632](https://github.com/NOAA-EMC/global-workflow/commit/ced16325b3de7e7fd013363e2c59478a9babe173)